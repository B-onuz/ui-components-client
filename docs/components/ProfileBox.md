# Pagination (Component)

Componente para perfil.

## Como utilizar

```js
import ProfileBox from 'ui-components/lib/components/ProfileBox'
```

```jsx
import React from 'react'
import ProfileBox from './ProfileBox'
import Button from '../Button'
import NavBar from '../NavBar'

export default {
  title: 'ProfileBox',
  component: ProfileBox,
};

export const ProfileBoxExample = () => (
  return (
    <ProfileBox balanceValue='2.300'/>
  )
)
```

## VariaÃ§Ãµes

### Variant

```jsx
<ProfileBox balanceValue="2.300" />
```
