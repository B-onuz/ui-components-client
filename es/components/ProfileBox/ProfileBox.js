"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports["default"] = void 0;

var _react = _interopRequireDefault(require("react"));

var _styledComponents = _interopRequireDefault(require("styled-components"));

var _styledSystem = require("styled-system");

var _Button = _interopRequireDefault(require("../Button"));

var _Typography = _interopRequireDefault(require("../Typography"));

var _this = void 0,
    _jsxFileName = "/Users/master/projects/bonuz/ui-components/src/components/ProfileBox/ProfileBox.js";

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { "default": obj }; }

function _extends() { _extends = Object.assign || function (target) { for (var i = 1; i < arguments.length; i++) { var source = arguments[i]; for (var key in source) { if (Object.prototype.hasOwnProperty.call(source, key)) { target[key] = source[key]; } } } return target; }; return _extends.apply(this, arguments); }

function _objectWithoutPropertiesLoose(source, excluded) { if (source == null) return {}; var target = {}; var sourceKeys = Object.keys(source); var key, i; for (i = 0; i < sourceKeys.length; i++) { key = sourceKeys[i]; if (excluded.indexOf(key) >= 0) continue; target[key] = source[key]; } return target; }

function _templateObject6() {
  var data = _taggedTemplateLiteralLoose(["\n  white-space: nowrap;\n  overflow: hidden;\n  text-overflow: ellipsis;\n  :hover {\n    color: ", ";\n    cursor: pointer;\n  }\n"]);

  _templateObject6 = function _templateObject6() {
    return data;
  };

  return data;
}

function _templateObject5() {
  var data = _taggedTemplateLiteralLoose(["\n  display: flex;\n  justify-content: flex-start;\n  align-items: center;\n"]);

  _templateObject5 = function _templateObject5() {
    return data;
  };

  return data;
}

function _templateObject4() {
  var data = _taggedTemplateLiteralLoose(["\n  height: ", ";\n  margin-right: 7px;\n  @media only screen and (max-width: 500px) {\n    max-height: 20px;\n    margin-right: 8px;\n    /* margin-bottom: 7px; */\n  }\n"]);

  _templateObject4 = function _templateObject4() {
    return data;
  };

  return data;
}

function _templateObject3() {
  var data = _taggedTemplateLiteralLoose(["\n  font-weight: bold;\n  font-size: 20px;\n  color: #000;\n  text-decoration: none !important;\n  :hover {\n    color: #ef4d5a;\n  }\n"]);

  _templateObject3 = function _templateObject3() {
    return data;
  };

  return data;
}

function _templateObject2() {
  var data = _taggedTemplateLiteralLoose(["\n  height: 100%;\n  display: flex;\n  justify-content: flex-start;\n  align-items: center;\n  @media only screen and (max-width: 500px) {\n    /* flex-direction: column; */\n    align-items: center;\n    /* justify-content: space-between; */\n    margin: 0 8px;\n  }\n"]);

  _templateObject2 = function _templateObject2() {
    return data;
  };

  return data;
}

function _templateObject() {
  var data = _taggedTemplateLiteralLoose(["\n  ", "\n  height: 100%;\n  display: flex;\n  justify-content: flex-end;\n  align-items: center;\n  @media only screen and (max-width: 500px) {\n    flex-direction: row;\n    align-items: center;\n    justify-content: center;\n    margin: 0 8px;\n  }\n"]);

  _templateObject = function _templateObject() {
    return data;
  };

  return data;
}

function _taggedTemplateLiteralLoose(strings, raw) { if (!raw) { raw = strings.slice(0); } strings.raw = raw; return strings; }

var StyledProfileBox = _styledComponents["default"].div(_templateObject(), _styledSystem.color);

var BnzBox = _styledComponents["default"].div(_templateObject2());

var StyledButton = (0, _styledComponents["default"])(_Button["default"])(_templateObject3());

var ProfileIcon = _styledComponents["default"].img(_templateObject4(), function (_ref) {
  var iconHeight = _ref.iconHeight;
  return iconHeight ? iconHeight : '100%';
});

var TextBox = _styledComponents["default"].div(_templateObject5());

var UserNameText = (0, _styledComponents["default"])(_Typography["default"])(_templateObject6(), function (_ref2) {
  var theme = _ref2.theme;
  return "" + theme.colors['gray'];
});

var ProfileBox = function ProfileBox(_ref3) {
  var children = _ref3.children,
      userName = _ref3.userName,
      balanceValue = _ref3.balanceValue,
      src = _ref3.src,
      iconHeight = _ref3.iconHeight,
      bnzOnClick = _ref3.bnzOnClick,
      userOnClick = _ref3.userOnClick,
      rest = _objectWithoutPropertiesLoose(_ref3, ["children", "userName", "balanceValue", "src", "iconHeight", "bnzOnClick", "userOnClick"]);

  return /*#__PURE__*/_react["default"].createElement(StyledProfileBox, _extends({}, rest, {
    __self: _this,
    __source: {
      fileName: _jsxFileName,
      lineNumber: 72,
      columnNumber: 5
    }
  }), !!userName && /*#__PURE__*/_react["default"].createElement(UserNameText, {
    color: "gray",
    m: 0,
    mr: 2,
    onClick: userOnClick,
    __self: _this,
    __source: {
      fileName: _jsxFileName,
      lineNumber: 74,
      columnNumber: 9
    }
  }, userName, " |"), (src || balanceValue) && /*#__PURE__*/_react["default"].createElement(BnzBox, {
    onClick: bnzOnClick,
    __self: _this,
    __source: {
      fileName: _jsxFileName,
      lineNumber: 79,
      columnNumber: 9
    }
  }, src && /*#__PURE__*/_react["default"].createElement(ProfileIcon, {
    alt: "bnz_icon",
    iconHeight: iconHeight,
    src: src,
    __self: _this,
    __source: {
      fileName: _jsxFileName,
      lineNumber: 80,
      columnNumber: 19
    }
  }), balanceValue && /*#__PURE__*/_react["default"].createElement(TextBox, {
    __self: _this,
    __source: {
      fileName: _jsxFileName,
      lineNumber: 82,
      columnNumber: 13
    }
  }, /*#__PURE__*/_react["default"].createElement(StyledButton, {
    linkButton: true,
    p: 0,
    __self: _this,
    __source: {
      fileName: _jsxFileName,
      lineNumber: 83,
      columnNumber: 15
    }
  }, balanceValue + " BNZ"))));
};

var _default = ProfileBox;
exports["default"] = _default;