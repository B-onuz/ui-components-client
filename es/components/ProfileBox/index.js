"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports["default"] = void 0;

var _ProfileBox = _interopRequireDefault(require("./ProfileBox"));

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { "default": obj }; }

var _default = _ProfileBox["default"];
exports["default"] = _default;