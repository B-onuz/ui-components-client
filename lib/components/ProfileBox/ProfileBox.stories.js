'use strict'

exports.__esModule = true
exports.ProfileBoxExample = exports['default'] = void 0

var _react = _interopRequireDefault(require('react'))

var _ProfileBox = _interopRequireDefault(require('./ProfileBox'))

var _Button = _interopRequireDefault(require('../Button'))

var _NavBar = _interopRequireDefault(require('../NavBar'))

var _bnz_icon = _interopRequireDefault(require('../../assets/bnz_icon.png'))

var _this = void 0,
  _jsxFileName = '/home/junior/Fontes/ui-components-client/src/components/ProfileBox/ProfileBox.stories.js'

function _interopRequireDefault(obj) {
  return obj && obj.__esModule ? obj : { default: obj }
}

var _default = {
  title: 'ProfileBox',
  component: _ProfileBox['default'],
}
exports['default'] = _default

var ProfileBoxExample = function ProfileBoxExample() {
  return /*#__PURE__*/ _react['default'].createElement(
    _react['default'].Fragment,
    null,
    /*#__PURE__*/ _react['default'].createElement(
      _NavBar['default'],
      {
        style: {
          marginBottom: 30,
        },
        __self: _this,
        __source: {
          fileName: _jsxFileName,
          lineNumber: 14,
          columnNumber: 5,
        },
      },
      /*#__PURE__*/ _react['default'].createElement(_ProfileBox['default'], {
        src: _bnz_icon['default'],
        userName: 'John Doe',
        balanceValue: '2.300',
        __self: _this,
        __source: {
          fileName: _jsxFileName,
          lineNumber: 15,
          columnNumber: 7,
        },
      })
    ),
    /*#__PURE__*/ _react['default'].createElement(
      _NavBar['default'],
      {
        style: {
          marginBottom: 30,
        },
        __self: _this,
        __source: {
          fileName: _jsxFileName,
          lineNumber: 17,
          columnNumber: 5,
        },
      },
      /*#__PURE__*/ _react['default'].createElement(_ProfileBox['default'], {
        src: _bnz_icon['default'],
        iconHeight: '34px',
        balanceValue: '2.300',
        __self: _this,
        __source: {
          fileName: _jsxFileName,
          lineNumber: 18,
          columnNumber: 7,
        },
      })
    ),
    /*#__PURE__*/ _react['default'].createElement(
      _NavBar['default'],
      {
        style: {
          marginBottom: 30,
        },
        __self: _this,
        __source: {
          fileName: _jsxFileName,
          lineNumber: 20,
          columnNumber: 5,
        },
      },
      /*#__PURE__*/ _react['default'].createElement(_ProfileBox['default'], {
        userName: 'John Doe',
        __self: _this,
        __source: {
          fileName: _jsxFileName,
          lineNumber: 21,
          columnNumber: 7,
        },
      })
    ),
    /*#__PURE__*/ _react['default'].createElement(
      _NavBar['default'],
      {
        style: {
          marginBottom: 30,
        },
        __self: _this,
        __source: {
          fileName: _jsxFileName,
          lineNumber: 23,
          columnNumber: 5,
        },
      },
      /*#__PURE__*/ _react['default'].createElement(_ProfileBox['default'], {
        src: _bnz_icon['default'],
        __self: _this,
        __source: {
          fileName: _jsxFileName,
          lineNumber: 24,
          columnNumber: 7,
        },
      })
    ),
    /*#__PURE__*/ _react['default'].createElement(
      _NavBar['default'],
      {
        style: {
          marginBottom: 30,
        },
        __self: _this,
        __source: {
          fileName: _jsxFileName,
          lineNumber: 26,
          columnNumber: 5,
        },
      },
      /*#__PURE__*/ _react['default'].createElement(_ProfileBox['default'], {
        balanceValue: '2.300',
        __self: _this,
        __source: {
          fileName: _jsxFileName,
          lineNumber: 27,
          columnNumber: 7,
        },
      })
    ),
    /*#__PURE__*/ _react['default'].createElement(
      _NavBar['default'],
      {
        style: {
          marginBottom: 30,
        },
        __self: _this,
        __source: {
          fileName: _jsxFileName,
          lineNumber: 29,
          columnNumber: 5,
        },
      },
      /*#__PURE__*/ _react['default'].createElement(_ProfileBox['default'], {
        userName: 'John Doe',
        balanceValue: '2.300',
        __self: _this,
        __source: {
          fileName: _jsxFileName,
          lineNumber: 30,
          columnNumber: 7,
        },
      })
    ),
    /*#__PURE__*/ _react['default'].createElement(
      _NavBar['default'],
      {
        __self: _this,
        __source: {
          fileName: _jsxFileName,
          lineNumber: 32,
          columnNumber: 5,
        },
      },
      /*#__PURE__*/ _react['default'].createElement(_ProfileBox['default'], {
        userName: 'John Doe',
        src: _bnz_icon['default'],
        __self: _this,
        __source: {
          fileName: _jsxFileName,
          lineNumber: 33,
          columnNumber: 7,
        },
      })
    )
  )
}

exports.ProfileBoxExample = ProfileBoxExample
