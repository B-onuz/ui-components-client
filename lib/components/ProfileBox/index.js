"use strict";

exports.__esModule = true;
exports["default"] = void 0;

var _ProfileBox = _interopRequireDefault(require("./ProfileBox"));

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { "default": obj }; }

var _default = _ProfileBox["default"];
exports["default"] = _default;
module.exports = exports.default;