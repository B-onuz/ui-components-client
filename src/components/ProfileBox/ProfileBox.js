import React from 'react'
import styled from 'styled-components'
import { color } from 'styled-system'
import Button from '../Button'
import Typography from '../Typography'

const StyledProfileBox = styled.div`
  ${color}
  height: 100%;
  display: flex;
  justify-content: flex-end;
  align-items: center;
  @media only screen and (max-width: 500px) {
    flex-direction: row;
    align-items: center;
    justify-content: center;
    margin: 0 8px;
  }
`

const BnzBox = styled.div`
  height: 100%;
  display: flex;
  justify-content: flex-start;
  align-items: center;
  @media only screen and (max-width: 500px) {
    /* flex-direction: column; */
    align-items: center;
    /* justify-content: space-between; */
    margin: 0 8px;
  }
`

const StyledButton = styled(Button)`
  font-weight: bold;
  font-size: 20px;
  color: #000;
  text-decoration: none !important;
  :hover {
    color: #ef4d5a;
  }
`

const ProfileIcon = styled.img`
  height: ${({ iconHeight }) => (iconHeight ? iconHeight : '100%')};
  margin-right: 7px;
  @media only screen and (max-width: 500px) {
    max-height: 20px;
    margin-right: 8px;
    /* margin-bottom: 7px; */
  }
`

const TextBox = styled.div`
  display: flex;
  justify-content: flex-start;
  align-items: center;
`

const UserNameText = styled(Typography)`
  white-space: nowrap;
  overflow: hidden;
  text-overflow: ellipsis;
  :hover {
    color: ${({ theme }) => `${theme.colors['gray']}`};
    cursor: pointer;
  }
`

const ProfileBox = ({ children, userName, balanceValue, src, iconHeight, bnzOnClick, userOnClick, ...rest }) => {
  return (
    <StyledProfileBox {...rest}>
      {!!userName && (
        <UserNameText color="gray" m={0} mr={2} onClick={userOnClick}>
          {userName} |
        </UserNameText>
      )}
      {(src || balanceValue) && (
        <BnzBox onClick={bnzOnClick}>
          {src && <ProfileIcon alt="bnz_icon" iconHeight={iconHeight} src={src} />}
          {balanceValue && (
            <TextBox>
              <StyledButton linkButton p={0}>{`${balanceValue} BNZ`}</StyledButton>
            </TextBox>
          )}
        </BnzBox>
      )}
    </StyledProfileBox>
  )
}

export default ProfileBox
