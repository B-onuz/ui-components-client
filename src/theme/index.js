const colors = {
  red: '#db2828',
  lightRed: '#ef4d5a',
  mediumRed: '#FF4F5D',
  darkRed: '#9c2543',
  fullDarkRed: '#550e33',
  orange: '#f2711c',
  yellow: '#fbbd08',
  olive: '#b5cc18',
  green: '#21BA45',
  teal: '#00b5ad',
  blue: '#2185d0',
  facebookBlue: '#3B5998',
  blueLink: '#4285F4',
  violet: '#6435c9',
  purple: '#a333c8',
  pink: '#e03997',
  brown: '#a5673f',
  grey: '#767676',
  textGrey: '#8D8D8D',
  lightGrey: '#6D6D6D',
  mediumGrey: '#838383',
  darkGrey: '#9A9A9A',
  boldGray: '#55555A',
  black: '#1b1c1d',
  white: '#fdfdfd',
  warning: '#7A4D05',
  success: '#1E561F',
  info: '#0E566C',
  danger: '#973937',
  cyan: '#1CB5AD',
  disabledGrey: '#ffffff1f',
  disabledGreyText: '#ffffff4d',
}

colors.primary = colors.lightRed
colors.secondary = colors.fullDarkRed
colors.default = '#e0e1e2'

const breakpoints = ['40em', '52em', '64em', '80em']

breakpoints.sm = breakpoints[0]
breakpoints.md = breakpoints[1]
breakpoints.lg = breakpoints[2]
breakpoints.xl = breakpoints[3]

const fontSizes = [12, 14, 16, 20, 24, 32]

export default {
  fontFamily: `'Lato', sans-serif`,
  colors,
  fontSizes,
  breakpoints,
}
