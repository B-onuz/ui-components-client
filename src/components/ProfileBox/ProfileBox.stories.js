import React from 'react'
import ProfileBox from './ProfileBox'
import Button from '../Button'
import NavBar from '../NavBar'
import bnz_icon from '../../assets/bnz_icon.png'

export default {
  title: 'ProfileBox',
  component: ProfileBox,
}

export const ProfileBoxExample = () => (
  <>
    <NavBar style={{ marginBottom: 30 }}>
      <ProfileBox src={bnz_icon} userName="John Doe" balanceValue="2.300" />
    </NavBar>
    <NavBar style={{ marginBottom: 30 }}>
      <ProfileBox src={bnz_icon} iconHeight={'34px'} balanceValue="2.300" />
    </NavBar>
    <NavBar style={{ marginBottom: 30 }}>
      <ProfileBox userName="John Doe" />
    </NavBar>
    <NavBar style={{ marginBottom: 30 }}>
      <ProfileBox src={bnz_icon} />
    </NavBar>
    <NavBar style={{ marginBottom: 30 }}>
      <ProfileBox balanceValue="2.300" />
    </NavBar>
    <NavBar style={{ marginBottom: 30 }}>
      <ProfileBox userName="John Doe" balanceValue="2.300" />
    </NavBar>
    <NavBar>
      <ProfileBox userName="John Doe" src={bnz_icon} />
    </NavBar>
  </>
)
