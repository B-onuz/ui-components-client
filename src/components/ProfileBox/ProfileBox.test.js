import React from 'react'
import ProfileBox from './ProfileBox.js'
import renderer from 'react-test-renderer'
import { ThemeProvider } from 'styled-components'
import theme from '../../theme'
import 'jest-styled-components'

test('Should render ProfileBox', () => {
  const component = renderer.create(
    <ThemeProvider theme={theme}>
      <ProfileBox />
    </ThemeProvider>
  )
  let tree = component.toJSON()
  expect(tree).toMatchSnapshot()
})
