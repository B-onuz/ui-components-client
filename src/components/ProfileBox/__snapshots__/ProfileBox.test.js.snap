// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Should render ProfileBox 1`] = `
.c0 {
  height: 100%;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: start;
  -webkit-justify-content: flex-start;
  -ms-flex-pack: start;
  justify-content: flex-start;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
}

@media only screen and (max-width:500px) {
  .c0 {
    -webkit-flex-direction: row;
    -ms-flex-direction: row;
    flex-direction: row;
    -webkit-align-items: center;
    -webkit-box-align: center;
    -ms-flex-align: center;
    align-items: center;
    -webkit-box-pack: center;
    -webkit-justify-content: center;
    -ms-flex-pack: center;
    justify-content: center;
    margin: 0 8px;
  }
}

<div
  className="c0"
/>
`;
